blocks/basic/test/qa_sources.cpp

```diff
@@ -164,6 +181,142 @@ const boost::ut::suite TagTests = [] {
 
         expect(eq(n_samples, static_cast<std::uint32_t>(sink.n_samples_produced))) << "Number of samples does not match";
     };
+
+    "FunctionGenerator ImChart test"_test = [] {
+        const std::size_t        N           = 128; // test points
+        double                   start_value = 10.;
+        double                   final_value = 20.;
+        std::vector<std::string> signals{ "Const", "LinearRamp", "ParabolicRamp", "CubicSpline", "ImpulseResponse" };
+        for (const auto &sig : signals) {
+            FunctionGenerator<double> funcGen{};
+            std::ignore = funcGen.settings().set({ { "signal_type", sig },
```

Comment:
as mentioned in the implemented code: we should consider replacing these string literals with either enum or default tag definitions that default convert to the required std::string key value. This would help readability (<-> public API/convention) and minimise typos.